File: src/main/java/de/thatsich/intellie/applied/intelligences/AppliedIntelligences.java
package de.thatsich.intellie.applied.intelligences;

import cpw.mods.fml.common.Mod;
import cpw.mods.fml.common.event.FMLPreInitializationEvent;

/**
 * @author thatsIch
 * @date 16.03.14.
 */
@Mod(
	modid = "appint",
	name = "AppliedIntelligences",
	version = "${version}",
	dependencies = "required-after:Forge@[7.0,);required-after:FML@[5.0.5,)" //;required-after:appliedenergistics2"
)
public class AppliedIntelligences
{
	public AppliedIntelligences ()
	{
		System.out.println( "CTOR" );
	}

	@Mod.EventHandler
	public void preInit ( FMLPreInitializationEvent event )
	{
		System.out.println( "POOOPS" );
	}
}

File: src/main/scala/de/thatsich/minecraft/intellie/applied/intelligences/AppliedIntelligences.scala
package de.thatsich.minecraft.intellie.applied.intelligences

import cpw.mods.fml.common.event.{FMLInitializationEvent, FMLPostInitializationEvent, FMLPreInitializationEvent}
import cpw.mods.fml.common.{Mod, SidedProxy}


/**
  *
  *
  * @author thatsIch
  * @since 05.04.2014.
  */
@Mod(
	modid = "appint",
	name = "AppliedIntelligences",
	version = "${version}",
	dependencies = "required-after:intellie",
	modLanguage = "scala"
)
object AppliedIntelligences extends IntelligencesAPI {
	final val id = "appint"
	final val name = "Applied Intelligences"
	final val version = "@version@"
	final val dependencies = "required-after:intellie"

	@SidedProxy(
		modId = AppliedIntelligences.id,
		clientSide = "de.thatsich.minecraft.intellie.applied.intelligences.proxy.IntelligencesClientProxy",
		serverSide = "de.thatsich.minecraft.intellie.applied.intelligences.proxy.IntelligencesServerProxy"
	)
	var proxy: IntelligencesCommonProxy = _

	@Mod.EventHandler
	def onPreInit(event: FMLPreInitializationEvent): Unit = this.proxy.onPreInit(event)

	@Mod.EventHandler
	def onInit(event: FMLInitializationEvent): Unit = this.proxy.onInit(event)

	@Mod.EventHandler
	def onPostInit(event: FMLPostInitializationEvent): Unit = this.proxy.onPostInit(event)
}
